modules/ipfs-cpinner-client/src/auth-manager.ts

```
@@ -0,0 +1,79 @@
+import { NO_DID, NO_SIGNER, ACCESS_TOKEN_KEY, REFRESH_TOKEN_KEY, NO_SERVICE_DID } from './constants'
+import { ClientKeyValueStorage, LoginResponse, Config } from './types'
+import axios from 'axios'
+import { createJWT } from 'did-jwt'
+
+export default (config: Config, storage: ClientKeyValueStorage) => {
+  const login = async (): Promise<LoginResponse> => {
+    const { did, signer, serviceUrl } = config
+    if (!did) throw new Error(NO_DID)
+    if (!signer) throw new Error(NO_SIGNER)
+
+    const tokens = await getChallenge()
+      .then(signChallenge)
+      .then(signature => axios.post(`${serviceUrl}/auth`, { response: signature }))
+      .then(res => res.status === 200 && res.data)
+      .catch(console.log)
```

This hides the error